scratch/evaluate_scheduler_qd_dense_scenario.cc

```diff
@@ -0,0 +1,825 @@
+  srcApp.Start (Seconds (simulationTime + 1));
+  srcApp.Stop (Seconds (simulationTime));
+  commPair.srcApp = srcApp.Get (0);
+  commPair.appDataRate = DataRate (appDataRate).GetBitRate ();
+
+  /* Install Simple TCP/UDP Server on the destination node */
+  PacketSinkHelper sinkHelper (socketType, InetSocketAddress (Ipv4Address::GetAny (), 9000 + appNumber));
+  ApplicationContainer sinkApp = sinkHelper.Install (dstNode);
+  commPair.packetSink = StaticCast<PacketSink> (sinkApp.Get (0));
+  sinkApp.Start (Seconds (0.0));
+
+  return commPair;
+}
+
+void
+SLSCompleted (Ptr<Parameters> parameters,
+              Mac48Address address, ChannelAccessPeriod accessPeriod,
+              BeamformingDirection beamformingDirection, bool isInitiatorTxss, bool isResponderTxss,
+              SECTOR_ID sectorId, ANTENNA_ID antennaId)
+{
+  string stationType;
+  if (parameters->wifiMac->GetTypeOfStation () == DMG_AP)
+    stationType = "DMG  AP=";    
+  else
+    stationType = "DMG STA=";
+
+  NS_LOG_DEBUG (stationType << parameters->wifiMac->GetAddress () << " completed SLS phase with " << address 
+                << ", antennaID=" << +antennaId << ", sectorID=" << +sectorId << ", accessPeriod=" << accessPeriod
+                << ", IsInitiator=" << (beamformingDirection == 0));
+    
+}
+
+void
+DataTransmissionIntervalStarted (Ptr<DmgStaWifiMac> wifiMac, Mac48Address address, Time)
+{
+  /* Avoiding beamforming training in a CBAP for the moment */
+  /*if (wifiMac->IsAssociated ())
+    {
+      uint16_t counter = biCounter[address];
+      counter++;
+      if (counter == biThreshold)
+        {
+          wifiMac->InitiateTxssCbap (wifiMac->GetBssid ());
+          counter = 0;
+        }
+      biCounter[address] = counter;
+    }
+    */
+}
+
+void
+MacRxOk (Ptr<DmgWifiMac> wifiMac, WifiMacType type, Ptr<const Packet> packet, 
+         Mac48Address address, double snrValue)
+{
+  macRxDataOk.at (wifiMac->GetAddress ()) += 1;
+}
+
+void
+MacTxDataFailed (Ptr<DmgWifiMac> wifiMac, Mac48Address address)
+{
+  macTxDataFailed.at (wifiMac->GetAddress ()) += 1;
+}
+
+void 
+MacTxOk (Ptr<DmgWifiMac> wifiMac, Mac48Address address)
+{
+  macTxDataOk.at (wifiMac->GetAddress ()) += 1;
+}
+
+int
+main (int argc, char *argv[])
+{
+  uint32_t bufferSize = 131072;                   /* TCP Send/Receive Buffer Size [bytes]. */
+  uint32_t queueSize = 1000;                      /* Wifi MAC Queue Size [packets]. */
+  string appDataRate = "300Mbps";                    /* Application data rate. */
+  bool frameCapture = false;                      /* Use a frame capture model. */
+  double frameCaptureMargin = 10;                 /* Frame capture margin [dB]. */
+  bool verbose = false;                           /* Print Logging Information. */
+  bool pcapTracing = false;                       /* Enable PCAP Tracing. */
+  uint16_t numSTAs = 8;                          /* The number of DMG STAs. */
```

Comment:
there are 10 STAs in the picture above (in the doc). Also, please write the max number of STAs supported